Tests/BSONTests/ExtendedJSONConversionTests.swift

```diff
@@ -251,6 +281,15 @@ open class ExtendedJSONConversionTestCase: BSONTestCase {
             .to(throwError(errorType: DecodingError.self))
     }
 
+    func testDatetime() throws {
+        // Canonical Success case
+        expect(try Date(fromExtJSON: ["$date": ["$numberLong": "500004"]], keyPath: []))
+            .to(equal(Date(msSinceEpoch: 500_004)))
+        // Relaxed Success case
+        expect(try Date(fromExtJSON: ["$date": "2012-12-24T12:15:30.501Z"], keyPath: []))
+            .to(equal(ExtendedJSONDecoder.extJSONDateFormatter.date(from: "2012-12-24T12:15:30.501Z")))
```

Comment:
this will basically always pass because its relying on the same formatter for parsing that the code is using. To make this test a little more meaningful, we should derive the expected date from another means like the milliseconds since epoch (which in this case for this date is `1356351330501`).

Reply:
done